api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1207 @@
+    get:
+      tags:
+        - schema-registry-api
+      description: |
+        Lists all of the enrichment strategies supported by this installation of Nakadi.
+
+        If the EventType creation is to have special enrichments (besides the default), one can consult over
+        this method the available possibilities.
+      responses:
+        '200':
+          description: Returns a list of all `EventEnrichmentStrategy` known to Nakadi
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/EventEnrichmentStrategy'
+
+
+  '/registry/partitioning-strategies':
+    get:
+      tags:
+        - schema-registry-api
+      description: |
+        Lists all of the partitioning strategies supported by this installation of Nakadi.
+
+        If the EventType creation is to have a specific partitioning (besides the default), one can
+        consult over this method the available possibilities.
+      responses:
+        '200':
+          description: Returns a list of all `EventPartitioningStrategy` known to Nakadi
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/PartitionResolutionStrategy'
+
+
+# ################################### #
+#                                     #
+#             Definitions             #
+#                                     #
+# ################################### #
+
+definitions:
+  Event:
+    type: object
+    description: |
+      **Note** The Event definition will be externalized in future versions of this document.
+
+      A basic payload of an Event. The actual schema is dependent on the information configured for the
+      EventType, as is its enforcement (see POST /event-types). Setting of metadata properties are
+      dependent on the configured enrichment as well.
+
+      For explanation on default configurations of validation and enrichment, see documentatin of
+      `EventType.type`.
+
+      For concrete examples of what will be enforced by Nakadi see the objects BusinessEvent and
+      DataChangeEvent below.
+    additionalProperties: true
+
+
+  EventMetadata:
+    type: object
+    description: |
+      Metadata for this Event.
+
+      Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon
+      reception, but they in general MIGHT be set by the client.
+    properties:
+      metadata:
+        type: object
+        properties:
+          eid:
+            description: |
+              Identifier of this Event.
+
+              Clients are allowed to generate this and this SHOULD be guaranteed to be unique from the
+              perspective of the producer. Consumers MIGHT use this value to assert uniqueness of reception
+              of the Event.
+
+              If not present upon reception, Nakadi will generate and fill this with a unique value.
+              (TBD: enforce UUID formatif set by the client?)
+            type: string
```

Comment:
@igoryankov yes, the `eid` is indeed though of as a tool to detect duplications.

We request and trust the producers to created unique `eid`s. @tfrauenstein even insists, that nakadi should not generate any ids by itself and just reject events that do not contain `eid` and I now also support him.

We plan to implement a service, that will be showing alerts in case some of the `event_type`s are getting `eid` duplications, but the semantics should be discussed when we start working on it.